dotcom-rendering/src/web/components/ShareIcons.tsx

```diff
@@ -105,7 +116,14 @@ export const ShareIcons = ({
 						rel="noreferrer"
 						data-ignore="global-link-styling"
 					>
-						<span css={iconStyles({ palette, size })}>
+						<span
+							css={[
+								iconStyles({ palette, size }),
+								format.design === ArticleDesign.LiveBlog &&
+									context === 'ArticleMeta' &&
+									articleMetaIconColor(palette),
```

Comment:
I think if we had one function to decide colour then we would be able to pass it `format` and `context` as props and let it make the choice. If you look around the codebase you'll see a lot of other style functions doing this, typically they are called `decideThing`, like decideWidth, decideSize, decidePosition, etc.

And I'm aware that I'm being a bit contradictory here with where logic should live, sometimes we have it in `css={[]}` and sometimes in the function! There's not really any rules, I think it's just about evaluating things on a case by case basis

Reply:
It's actually really good that you're explaining your views with patience and I appreciate it very much. It's giving me a better perspective on how to write better code :)